### PR TITLE
Add EstablishmentModal component

### PR DESCRIPTION
**Pull request by jfplataroti**
Added `EstablishmentModal` component used in `DeliveryTabs` used for mobile screens.

Not a lot of changes, it follows the same behaviour than the `AddressModal` component. Also, it uses the `useEstablishment` hook used in `EstablishmentField` to load the establishment options.

Modal with geoLocation enabled:

establishment picker

Modal with geoLocation disabled:

establishment-picker-modal-geo-disabled

### CODE REVIEW

File: packages/web/components/deliveryTabs.tsx [hunk 1]
```diff
@@ -9,22 +9,30 @@ import ClockIcon from '../public/images/clock.svg';
...
 
 export const DeliveryTabs: React.FunctionComponent = () => {
   const { t } = useTranslation();
-  const { value, setMode, setAddress } = useDeliveryOptions();
+  const { value, setMode, setAddress, setEstablishment } = useDeliveryOptions();
   const [showDeliveryModal, setShowDeliveryModal] = useState(false);
+  const [showEstablishmentModal, setShowEstablishmentModal] = useState(false);
 
-  const onAddressFieldSelected = () => setShowDeliveryModal(true);
+  const openAddressModal = () => setShowDeliveryModal(true);
```
Comment (djoyal) [inline packages/web/components/deliveryTabs.tsx:21]: 👍

File: packages/web/components/establishmentModal/establishmentModal.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  onSelected,
+  onClosed,
+}) => {
+  const { establishments } = useEstablishment();
+  const { t } = useTranslation();
+
+  return (
+    <Layer onClickOutside={onClosed}>
+      <Box pad="medium" margin={{ top: 'small' }} width="large">
+        <Button fill={false} plain={true}>
```
Comment (djoyal) [inline packages/web/components/establishmentModal/establishmentModal.tsx:27]: You can actually condense this a bit more by embedding the icon directly in the button's props (E.G. `<Button icon={<Close />} />` :)

File: packages/web/components/establishmentModal/suggestionListItem.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+import storeIcon from '../../public/images/store-icon.svg';
+
+interface Props {
+  item: Establishment;
+  selected?: boolean;
+  onSelected?: (establishment: Establishment) => void;
+}
+
+const formateDistance = (distance: number) =>
```
Comment (djoyal) [inline packages/web/components/establishmentModal/suggestionListItem.tsx:14]: Extra `e` in format

File: packages/web/components/establishmentModal/suggestionListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  const { streetAddress, streetNumber, city, state, postalCode } = item.address;
+  const mainAddressText = `${streetNumber} ${streetAddress}, ${city},`;
+  const secondaryAddressText = `${state}, ${postalCode}`;
+  const distance = item.distance ? `${formateDistance(item.distance)} km` : '';
+  const border = selected
+    ? { border: `1px solid ${theme.global.colors.brand}` }
+    : {};
+  const buttonStyle = { borderRadius: '12px', ...border };
+
+  const onClickHander = () => onSelected && onSelected(item);
```
Comment (djoyal) [inline packages/web/components/establishmentModal/suggestionListItem.tsx:31]: `onClickHandler` doesn't tell me much about what this function does, could you use a more descriptive name please?

**Review by djoyal — APPROVED** [review]
The component looks great! There's a few small comments but nothing blocking. Nice work!

File: packages/web/components/establishmentModal/establishmentModal.tsx (hunk 2 above)
Comment (jfplataroti) [reply packages/web/components/establishmentModal/establishmentModal.tsx:27]: nice! I'll update it.

File: packages/web/components/establishmentModal/suggestionListItem.tsx (hunk 4 above)
Comment (jfplataroti) [reply packages/web/components/establishmentModal/suggestionListItem.tsx:31]: yes, sorry it doesn't follow the naming convention agreed in the previous PR.

Comment (jfplataroti) [reply packages/web/components/establishmentModal/suggestionListItem.tsx]: updated :)

**Comment by jfplataroti** [comment]
Hey @djoyal , I added the suggestions (the naming convention for the event handlers and simplifying the button using the icon attribute).

I added another commit to remove some code duplication when the address and distance were formatted for the establishment suggestions.

**Review by djoyal — APPROVED** [review]
Thanks for cleaning up the naming. I also really like moving the `formatAddress` function to the `lib/helpers`! 🙌

**Pull request merged**